### PR TITLE
refactor to common/download/upload scripts

### PR DESCRIPTION
**Pull request by aparcar**
docker-download.sh - Downloads FILE_DOWNLOAD from FILE_HOST and verifies

docker-upload.sh - Uploads the created docker images to docker.io

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by aparcar** [comment]
@oxr463 I'd be happy if you could test this locally!

**Comment by oxr463** [comment]
Does CircleCI not run on PRs?

File: docker-download.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
```
Comment (oxr463) [inline docker-download.sh]: Couldn't we just put these on one line?

```sh
set -ex
```

File: docker-download.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+set -x
+
+export FILE_HOST="${FILE_HOST:-downloads.openwrt.org}"
+
+curl "https://$FILE_HOST/$DOWNLOAD_PATH/sha256sums" -fs -o sha256sums
+curl "https://$FILE_HOST/$DOWNLOAD_PATH/sha256sums.asc" -fs -o sha256sums.asc || true
+curl "https://$FILE_HOST/$DOWNLOAD_PATH/sha256sums.sig" -fs -o sha256sums.sig || true
+if [ ! -f sha256sums.asc ] && [ ! -f sha256sums.sig ]; then
+    echo_red "Missing sha256sums signature files"
```
Comment (oxr463) [inline docker-download.sh]: Where does this `echo_red` come from?

File: docker-rootfs.sh [hunk 3]
```diff
@@ -1,44 +1,29 @@
 #!/bin/sh
 
+set -e
+set -x
```
Comment (oxr463) [inline docker-rootfs.sh]: See previous comment.

**Comment by aparcar** [comment]
Not just yet, I have a PR ready for that but it requires a bit more finishing

**Comment by aparcar** [comment]
@oxr463 seems to work, please try it locally

**Pull request merged**